File: core/src/main/java/com/mcreater/amclcore/exceptions/oauth/OAuthMinecraftNameChangeNotAllowedException.java
package com.mcreater.amclcore.exceptions.oauth;

import lombok.AllArgsConstructor;

import java.io.IOException;

import static com.mcreater.amclcore.i18n.I18NManager.translatable;

@AllArgsConstructor
public class OAuthMinecraftNameChangeNotAllowedException extends IOException {
    private final String lastChange;
    private final String nextChange;
    private final long deltaTime;

    public String toString() {
        return translatable("core.exceptions.oauth.name_change_not_allowed", lastChange, nextChange, deltaTime).getText();
    }
}

File: core/src/main/java/com/mcreater/amclcore/exceptions/oauth/OAuthMinecraftNameChangeNotAllowedException.kt
package com.mcreater.amclcore.exceptions.oauth

import com.mcreater.amclcore.i18n.I18NManager.Companion.translatable
import java.io.IOException


data class OAuthMinecraftNameChangeNotAllowedException(
    val lastChange: String? = null,
    val nextChange: String? = null,
    val deltaTime: Long = 0
) : IOException() {
    override fun toString(): String {
        return translatable(
            "core.exceptions.oauth.name_change_not_allowed",
            lastChange!!, nextChange!!, deltaTime
        ).text!!
    }
}
